server/src/main/java/org/apache/druid/server/coordinator/CuratorLoadQueuePeon.java

```diff
@@ -219,199 +220,182 @@ public void unmarkSegmentToDrop(DataSegment dataSegment)
     segmentsMarkedToDrop.remove(dataSegment);
   }
 
-  private void processSegmentChangeRequest()
+  private class SegmentChangeProcessor implements Runnable
   {
-    if (currentlyProcessing != null) {
-      log.debug(
-          "Server[%s] skipping processSegmentChangeRequest because something is currently loading[%s].",
-          basePath,
-          currentlyProcessing.getSegmentId()
-      );
+    private final SegmentHolder segmentHolder;
 
-      return;
-    }
-
-    if (!segmentsToDrop.isEmpty()) {
-      currentlyProcessing = segmentsToDrop.firstEntry().getValue();
-      log.debug("Server[%s] dropping [%s]", basePath, currentlyProcessing.getSegmentId());
-    } else if (!segmentsToLoad.isEmpty()) {
-      currentlyProcessing = segmentsToLoad.firstEntry().getValue();
-      log.debug("Server[%s] loading [%s]", basePath, currentlyProcessing.getSegmentId());
-    } else {
-      return;
+    private SegmentChangeProcessor(SegmentHolder segmentHolder)
+    {
+      this.segmentHolder = segmentHolder;
     }
 
-    try {
-      final String path = ZKPaths.makePath(basePath, currentlyProcessing.getSegmentId().toString());
-      final byte[] payload = jsonMapper.writeValueAsBytes(currentlyProcessing.getChangeRequest());
-      curator.create().withMode(CreateMode.EPHEMERAL).forPath(path, payload);
-
-      processingExecutor.schedule(
-          () -> {
-            try {
-              if (curator.checkExists().forPath(path) != null) {
-                failAssign(new ISE("%s was never removed! Failing this operation!", path));
+    @Override
+    public void run()
+    {
+      try {
+        final String path = ZKPaths.makePath(basePath, segmentHolder.getSegmentIdentifier());
+        final byte[] payload = jsonMapper.writeValueAsBytes(segmentHolder.getChangeRequest());
+        curator.create().withMode(CreateMode.EPHEMERAL).forPath(path, payload);
+        log.debug(
+            "ZKNode created for server to [%s] %s [%s]",
+            basePath,
+            segmentHolder.getType() == LOAD ? "load" : "drop",
+            segmentHolder.getSegmentIdentifier()
+        );
+        final ScheduledFuture<?> future = monitorNodeRemovedExecutor.schedule(
```

Comment:
Please extract this block as a method to reduce the size of `run()`.

Reply:
@gianm could you please stop following every PR that I review and comment on what you think is minor and what you think is not?

Reply:
I agree with @gianm . Comments like these, if need be, should be made towards the start of  the review. It adds unnecessary friction and discourages contributors in general. Refactoring like this doesn't contribute anything to the functionality, increases the scope of changes and makes figuring out issues in the code through reviews difficult. Having said that, I appreciate issues you identified and some other changes you have suggested @leventov .

Reply:
> Refactoring like this doesn't contribute anything to the functionality, increases the scope of changes and makes figuring out issues in the code through reviews difficult.

These points that you brought are not special to the proposed refactoring - they are pretty much by definition of *any* refactoring. So does this mean that refactoring shouldn't be done at all? (Making refactoring in separate PRs is not a solution as well, at least because in practice, nobody happens to do that in Druid, or do very rarely.)

> Comments like these, if need be, should be made towards the start of the review

I agree with the point that *all* comments should be preferably made towards the start of the review, rather than in small batches. It's a place for improvement for me, but we have what we have here already.

So could you please extract the method?

Reply:
Done